webots_ros2_core/webots_ros2_core/joint_state_publisher.py

```diff
@@ -46,7 +46,8 @@ def __init__(self, robot, jointPrefix, node):
         for i in range(robot.getNumberOfDevices()):
             device = robot.getDeviceByIndex(i)
             if device.getNodeType() == Node.POSITION_SENSOR:
```

Comment:
```suggestion
            if device.getNodeType() == Node.Motor:
```

Reply:
We only need the motor name, the rest still needs the sensor. So I think my solution is the simpler one. We could switch to select for motor, but then need to assign the sensor by `sensor = device.getPositionSensor()`

Reply:
Oh ok I understand, in that case, we should probably change the name to:
```Python
name = motor.getName() if motor is not None else name = device.getName()
```
Just to handle the case where there is no motor on this joint.

Reply:
sounds good to me, can you add the change?